lib/Platform/Intl/PlatformIntlApple.mm

```diff
@@ -337,48 +490,1338 @@ LocaleMatch lookupMatcher(
+    // k = h24 = 1-24
+    if (impl_->HourCycle == u"h12") {
+      switch (formatDate(impl_->Hour)) {
+        case eNumeric: {
+          [customFormattedDate appendString:@"h"];
+          break;
+        }
+        case eTwoDigit: {
+          [customFormattedDate appendString:@"hh"];
+          break;
+        }
+        default: {
+          [customFormattedDate appendString:@"hh"];
+        }
+      }
+    } else if (impl_->HourCycle == u"h24") {
+      switch (formatDate(impl_->Hour)) {
+        case eNumeric: {
+          [customFormattedDate appendString:@"k"];
+          break;
+        }
+        case eTwoDigit: {
+          [customFormattedDate appendString:@"kk"];
+          break;
+        }
+        default: {
+          [customFormattedDate appendString:@"kk"];
+        }
+      }
+    } else if (impl_->HourCycle == u"h11") {
+      switch (formatDate(impl_->Hour)) {
+        case eNumeric: {
+          [customFormattedDate appendString:@"K"];
+          break;
+        }
+        case eTwoDigit: {
+          [customFormattedDate appendString:@"KK"];
+          break;
+        }
+        default: {
+          [customFormattedDate appendString:@"KK"];
+        }
+      }
+    } else { // h23
+      switch (formatDate(impl_->Hour)) {
+        case eNumeric: {
+          [customFormattedDate appendString:@"H"];
+          break;
+        }
+        case eTwoDigit: {
+          [customFormattedDate appendString:@"HH"];
+          break;
+        }
+        default: {
+          [customFormattedDate appendString:@"HH"];
+        }
+      }
+    }
+  }
+  if (impl_->Minute != u"") {
+    switch (formatDate(impl_->Minute)) {
+      case eNumeric: {
+        [customFormattedDate appendString:@"m"];
+        break;
+      }
+      case eTwoDigit: {
+        [customFormattedDate appendString:@"mm"];
+        break;
+      }
+      default: {
+        [customFormattedDate appendString:@"m"];
+      }
+    }
+  }
+  if (impl_->Second != u"") {
+    switch (formatDate(impl_->Second)) {
+      case eNumeric: {
+        [customFormattedDate appendString:@"s"];
+        break;
+      }
+      case eTwoDigit: {
+        [customFormattedDate appendString:@"ss"];
+        break;
+      }
+      default: {
+        [customFormattedDate appendString:@"s"];
+      }
+    }
+  }
+  if (impl_->FractionalSecondDigits != NAN) {
+    // This currently outputs to 3 digits only with the date?
+    switch (impl_->FractionalSecondDigits) {
+      case 1: {
+        [customFormattedDate appendString:@"S"];
+      }
+      case 2: {
+        [customFormattedDate appendString:@"SS"];
+      }
+      case 3: {
+        [customFormattedDate appendString:@"SSS"];
+      }
+    }
+  }
+  // Not supported - dayPeriod (at night/in the morning)
+  // Set the custom date from all the concatonated NSStrings (locale will
+  // automatically seperate the order) Only set a template format if it isn't
+  // empty
+  if (![customFormattedDate isEqual:@""]) {
```

Comment:
It is more common to check string length in Objective-C. So this condition would read like `customFormattedDate.length > 0`